rustyms-py/src/lib.rs

```diff
@@ -0,0 +1,1262 @@
+    /// str
+    ///
+    #[getter]
+    fn ion(&self) -> String {
+        self.0.ion.to_string() // TODO: Return as exposed Fragment type
+    }
+
+    /// The peptide this fragment comes from, saved as the index into the list of peptides in the overarching crate::ComplexPeptide struct.
+    ///
+    /// Returns
+    /// -------
+    /// int
+    ///
+    #[getter]
+    fn peptide_index(&self) -> usize {
+        self.0.peptide_index
+    }
+
+    /// Any neutral losses applied.
+    ///
+    /// Returns
+    /// -------
+    /// str | None
+    ///
+    #[getter]
+    fn neutral_loss(&self) -> Option<String> {
+        match &self.0.neutral_loss {
+            Some(nl) => Some(nl.to_string()),
+            None => None,
+        }
+    }
+
+    /// Additional description for humans.
+    ///
+    /// Returns
+    /// -------
+    /// str
+    ///
+    #[getter]
+    fn label(&self) -> String {
+        self.0.label.clone()
+    }
+}
+
+/// One block in a sequence meaning an amino acid and its accompanying modifications.
+#[pyclass]
+pub struct SequenceElement(rustyms::SequenceElement);
+
+#[pymethods]
+impl SequenceElement {
+    fn __repr__(&self) -> String {
+        format!("SequenceElement(amino_acid='{}', modifications='{:?}', possible_modifications='{:?}', ambiguous='{:?}')", self.aminoacid(), self.modifications(), self.possible_modifications(), self.ambiguous())
+    }
+
+    /// The amino acid.
+    ///
+    /// Returns
+    /// -------
+    /// AminoAcid
+    ///
+    #[getter]
+    fn aminoacid(&self) -> AminoAcid {
+        AminoAcid(self.0.aminoacid)
+    }
+
+    /// All present modifications.
+    ///
+    /// Returns
+    /// -------
+    /// list[Modification]
+    ///
+    #[getter]
+    fn modifications(&self) -> Vec<Modification> {
+        self.0
+            .modifications
+            .iter()
+            .map(|m| Modification(m.clone()))
+            .collect()
+    }
+
+    /// All ambiguous modifications (could be placed here or on another position).
+    ///
+    /// Returns
+    /// -------
+    /// list[AmbiguousModification]
+    ///
+    #[getter]
+    fn possible_modifications(&self) -> Vec<AmbiguousModification> {
+        self.0
+            .possible_modifications
+            .iter()
+            .map(|m| AmbiguousModification(m.clone()))
+            .collect()
+    }
+
+    /// If this amino acid is part of an ambiguous sequence group `(QA)?` in ProForma
+    ///
+    /// Returns
+    /// -------
+    /// int | None
+    ///
+    #[getter]
+    fn ambiguous(&self) -> Option<usize> {
+        self.0.ambiguous
+    }
+
+    /// Get the molecular formula for this position (unless it is B/Z) with the selected ambiguous modifications, without any global isotype modifications.
+    ///
+    /// Parameters
+    /// ----------
+    /// selected_ambiguous : int
+    ///
+    /// Returns
+    /// -------
+    /// MolecularFormula
+    ///
+    fn formula(&self, selected_ambiguous: usize) -> MolecularFormula {
```

Comment:
Did you also provide methods for the `formulas` equivalents? Those are the only ones I use internally, with these exposed for people that know their stuff does not contain B/Z.

Reply:
I'm not entirely following here. What do you mean with `formulas` equivalents?

Reply:
The methods called `formulas` that returns a `Multi<Mass>`.  There are `*_greedy` and `*_all` equivalents.

Reply:
Whoops. Was not using the latest rustyms from the repo, but using 0.7.0 instead. Had not yet noticed all the changes.